src/Security/RolesBuilder/AdminRolesBuilder.php

```diff
@@ -76,27 +78,65 @@ public function addExcludeAdmin(string $exclude): void
 
     public function getRoles(?string $domain = null): array
     {
+        $adminServiceCodes = array_diff($this->pool->getAdminServiceCodes(), $this->excludeAdmins);
+
+        // get groups and admins sort by config
         $adminRoles = [];
-        foreach ($this->pool->getAdminServiceCodes() as $code) {
-            if (\in_array($code, $this->excludeAdmins, true)) {
-                continue;
-            }
+        foreach ($this->pool->getAdminGroups() as $groupCode => $group) {
+            foreach ($group['items'] as $item) {
+                if (!isset($item['admin'])) {
+                    continue;
+                }
+
+                $key = array_search($item['admin'], $adminServiceCodes, true);
+                if (false === $key) {
+                    continue;
+                }
+                unset($adminServiceCodes[$key]);
 
-            $admin = $this->pool->getInstance($code);
-            $securityHandler = $admin->getSecurityHandler();
-            $baseRole = $securityHandler->getBaseRole($admin);
-            foreach (array_keys($admin->getSecurityInformation()) as $key) {
-                $role = sprintf($baseRole, $key);
-                $adminRoles[$role] = [
-                    'role' => $role,
-                    'label' => $key,
-                    'role_translated' => $this->translateRole($role, $domain),
-                    'is_granted' => $this->isMaster($admin) || $this->authorizationChecker->isGranted($role),
-                    'admin_label' => $admin->getTranslator()->trans($admin->getLabel() ?? ''),
-                ];
+                $groupLabelTranslated = $this->translator->trans($group['label'], [], $group['translation_domain']);
+
+                $adminRoles = array_merge($adminRoles, $this->getAdminRolesByAdminCode($item['admin'], $domain, $groupLabelTranslated, $groupCode));
             }
         }
 
+        // admin with config "show_in_dashboard" set "false" does not have group
+        $defaultGroupLabelTranslated = $this->translator->trans('_', [], $domain);
+        $defaultGroupCode = '_';
+        foreach ($adminServiceCodes as $code) {
+            $adminRoles = array_merge($adminRoles, $this->getAdminRolesByAdminCode($code, $domain, $defaultGroupLabelTranslated, $defaultGroupCode));
+        }
+
+        return $adminRoles;
+    }
+
+    /**
+     * @return array<string, array<string, string|bool>>
+     *
+     * @phpstan-return array<string, Role>
+     */
+    private function getAdminRolesByAdminCode(string $code, ?string $domain = null, string $groupLabelTranslated = '_', string $groupCode = '_'): array
```

Comment:
The `_` default seems weird.

Reply:
"groupLabelTranslated" empty string is better? groupCode replace to "default" ?

Reply:
Yeah, I think empty string will be better.